source/simulator/freesurface.cc

```diff
@@ -97,6 +116,25 @@ namespace aspect
       else
         AssertThrow(false, ExcMessage("The surface velocity projection must be ``normal'' or ``vertical''."));
 
+      try
+        {
+          const std::vector<types::boundary_id> x_additional_tangential_mesh_boundary_indicators
+            = sim.geometry_model->translate_symbolic_boundary_names_to_ids(Utilities::split_string_list
+                                                                      (prm.get ("Additional tangential mesh velocity boundary indicators")));
+
+          tangential_mesh_boundary_indicators = sim.parameters.tangential_velocity_boundary_indicators;
+          tangential_mesh_boundary_indicators.insert(x_additional_tangential_mesh_boundary_indicators.begin(),
+                                                     x_additional_tangential_mesh_boundary_indicators.end());
+        }
+      catch (const std::string &error)
+        {
+          AssertThrow (false, ExcMessage ("While parsing the entry <Free surface/Additional tangential "
+                                          "mesh velocity boundary indicators>, there was an error. Maybe you "
+                                          "specified one of the <Model settings/Tangential velocity boundary indicators> "
+                                          "again, which is unncecessary? Specifically, "
+                                          "the conversion function complained as follows: "
+                                          + error));
```

Comment:
I am not sure that this will catch the error that you want it to. As I understand this, won't it 

1. Catch an error if `translate_symbolic_boundary_names_to_ids` fails, which is not what is described in the error message, and 
2. Not complain if there is a repeated boundary indicator for the `tagential_velocity_boundary_indicators` object? I don't think that std::set has a problem with re-inserting values.